salsa_webqa/salsa_runner.py

```diff
@@ -115,10 +117,27 @@ def run_tests(self):
             sys.exit('The runner cannot be executed directly.'
                      ' You need to import it within project specific runner. Session terminated.')
         else:
-            if self.driver_name.lower() == 'browserstack':
-                test_status = self.run_on_browserstack()
-            else:
-                test_status = self.run_locally()
+            self.cleanup_results()
+            if self.reporting == 'simple' or self.driver_name is None:
+                test_status = self.run_non_selenium()
+            elif self.reporting == 'all':
+                if self.driver_name.lower() == 'browserstack':
+                    test_status_selenium = self.run_on_browserstack()
+                    test_status_simple = self.run_non_selenium()
+                else:
+                    test_status_selenium = self.run_locally()
+                    test_status_simple = self.run_non_selenium()
```

Comment:
`test_status_simple = self.run_non_selenium()` is in both condition branches = therefore it can be taken out and put after the `if` `else` statement.